lib/index.js

```diff
@@ -170,16 +170,107 @@ class VideoAsset extends BaseAsset {
 
 exports.VideoAsset = VideoAsset;
 
+function _ensure_cheerio(value) {
+    if (typeof value === 'string')
+        return cheerio.load(value);
+    return value;
+}
+
 class NewsArticle extends BaseAsset {
     constructor() {
         super();
         this._object_type = "ArticleObject";
         this._content_type = "text/html";
     }
 
-    set_document(value) { this._document = cheerio.load(value); }
+    // Takes a Date or something that a Date can be constructed out of
+    set_date_published(value) {
+        if (value instanceof Date)
+            this._date_published = value;
+        else
+            this._date_published = new Date(value);
+    }
+
+    // Takes a string or array of strings
+    set_authors(value) {
+        if (value instanceof Array)
+            this._authors = value;
+        else
+            this._authors = [value];
+    }
+
+    set_main_image(asset, caption) {
+        this._main_image = asset;
+        this._main_caption = _ensure_cheerio(caption);
```

Comment:
`s/_main_caption/_main_image_caption/`. If it's the generqal caption, we should have a different setter for it.